File: python/grade-school/grade_school.py
from collections import defaultdict


class School(object):
    def __init__(self):
        self._db = defaultdict(set)

    def grade(self, n):
        return sorted(list(self._db[n])) or []

    def add_student(self, name, grade):
        self._db[grade].add(name)

    def roster(self):
        return [name
                for k, v in sorted(self._db.items())
                for name in sorted(v)
                ]

File: javascript/grade-school/grade-school.js
class School {
    constructor(name) {
        this.name = name;
        this.db = {};
    }

    roster() {
        return this.db;
    }

    grade(n) {
        return this.db[n] || [];
    }

    add(student, n) {
        if (this.db[n] === undefined) {
            this.db[n] = [];
        }
        this.db[n].push(student);
        this.db[n].sort();
    }
}

module.exports = School;

if (require.main === module) {
    let s = new School("Leader of the New School");
    s.add("Busta Rhymes", 1);
    s.add("Charlie Brown", 1);
    s.add("Dinco D", 1);
    s.add("Cut Monitor Milo", 1);
    console.log(s);
}
